src/components/SignUpForm/index.tsx

```diff
@@ -48,16 +48,16 @@ const SignUpForm: FC<SignUpProps> = ({ onSubmit }) => {
         <Text>{t("common:words.or")}</Text>
       </div>
 
-      <Form className={styles.form}>
+      <Form className={styles.form} onFinish={handleSubmit}>
```

Comment:
prop onfinish is not needed, the handler is already on the button. in addition, the handleSubmit function expects a callback to be passed to it